core/src/services/dropbox/core.rs

```diff
@@ -337,11 +337,14 @@ impl DropboxCore {
                     (path, Ok(RpDelete::default().into()))
                 }
                 "failure" => {
-                    let error = entry.error.expect("error should be present");
-                    let err = Error::new(
-                        ErrorKind::Unexpected,
-                        &format!("delete failed with error {}", error.error_summary),
-                    );
+                    let err = if let Some(error) = entry.error {
```

Comment:
It's a bit odd that dropbox returns a failure without an error. Do you know the actual response from dropbox services?

Reply:
Got it. Seems we need to parse the error response from dropbox correctly.